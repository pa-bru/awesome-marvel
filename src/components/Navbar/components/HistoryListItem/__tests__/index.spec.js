import React from 'react'
import { shallow } from 'enzyme'

import { HistoryListItem } from '../'

// Given
const props = {
  id: 1,
  name: 'Thor',
  onClick: jest.fn(),
  history: {
    push: jest.fn(),
  },
}

describe('HistoryListItem', () => {
  it('should render a ListItem correctly', () => {
    // When
    const tree = shallow(<HistoryListItem {...props} />)

    // Then
    expect(tree).toMatchSnapshot()
  })

  it('should call history.push on click on the list item', () => {
    // Given
    const tree = shallow(<HistoryListItem {...props} />)

    // When
    tree.simulate('click')

    // Then
    expect(props.history.push).toHaveBeenCalledWith(`/characters/${props.id}`)
  })
})
