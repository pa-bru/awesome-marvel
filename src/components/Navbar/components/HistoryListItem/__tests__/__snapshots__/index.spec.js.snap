// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryListItem should render a ListItem correctly 1`] = `
<WithStyles(ListItem)
  button={true}
  dense={true}
  onClick={[Function]}
>
  <WithStyles(Avatar)
    aria-label="Recipe"
    src="."
  />
  <WithStyles(ListItemText)
    primary="Thor"
    secondary="0 comics - 0 series - 0 events"
  />
  <WithStyles(ListItemSecondaryAction)>
    <WithStyles(IconButton)
      aria-label="Delete"
      onClick={[Function]}
    >
      <pure(Delete) />
    </WithStyles(IconButton)>
  </WithStyles(ListItemSecondaryAction)>
</WithStyles(ListItem)>
`;
